tox.ini

```
@@ -1,37 +1,78 @@
-# Copyright 2021 Canonical Ltd.
+# Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
 
 [flake8]
 max-line-length = 100
-ignore = W503
-copyright-check = True
-copyright-author = Canonical Ltd.
-select = E,F,W,C
 
 [tox]
 skipsdist = True
+skip_missing_interpreters = True
+envlist = fmt, lint, unit, integration
+
+[vars]
+all_path = {[vars]src_path} {[vars]tst_path}
+src_path = {toxinidir}/src/
+tst_path = {toxinidir}/tests/
 
 [testenv]
-passenv = HOME
+passenv =
+    PYTHONPATH
+    CHARM_BUILD_DIR
+    MODEL_SETTINGS
+    KUBECONFIG
 setenv =
-    PYTHONPATH={toxinidir}/src:{toxinidir}/lib
+    PYTHONPATH = {toxinidir}:{toxinidir}/lib:{[vars]src_path}
+    PYTHONBREAKPOINT=ipdb.set_trace
+    PY_COLORS=1
+
+[testenv:update-requirements]
+allowlist_externals =
+    bash
+    find
+    pip-compile
+    xargs
+commands = 
+; uses 'bash -c' because piping didn't work in regular tox commands
+  bash -c \'find . -type f -name "requirements*.in" | xargs --replace=\{\} pip-compile --resolver=backtracking \{\}'
 deps =
-    -rtest-requirements.txt
-    -rrequirements.txt
+    -r requirements-update-packages.txt
```

To avoid the chicken-and-egg problem with this (eg: how this tox env requires the requirements-update-packages.txt that it itself produces), what do you say to (for this env only) explicitly writing the dependencies here?  Like:

```suggestion
    pip-tools
```
?

---

Agree